bundles/org.openhab.persistence.jdbc/src/main/java/org/openhab/persistence/jdbc/internal/utils/StringUtilsExt.java

```diff
@@ -161,24 +161,22 @@ public static Properties parseJdbcURL(String url, @Nullable Properties def) {
             props.put("pathQuery", dbURI.getQuery());
         }
 
-        String path = "";
-        if (dbURI.getPath() != null) {
-            String gp = dbURI.getPath();
-            String st = "/";
-            if (gp.indexOf("/") <= 1) {
-                if (substrPos(gp, st).size() > 1) {
-                    path = stringBeforeLastSubstr(gp, st) + st;
+        String pathURI = dbURI.getPath();
+        if (pathURI != null) {
+            String path = "";
+            if ((pathURI.indexOf("/") >= 0) && (pathURI.indexOf("/") <= 1)) {
+                if (pathURI.contains("/")) {
```

Comment:
Condition of `pathURI.indexOf("/")` in line 167 must either be 0 or 1 to be true.  conclusion `/` exists.
Then condition in line 168 will always result true.
Line 171 will never be called.

Reply:
Thanks for noticing.